projects/igniteui-angular/src/lib/services/transaction/transaction.ts

```diff
@@ -26,6 +35,11 @@ export interface State {
     type: TransactionType;
 }
 
+export interface IStateUpdateEvent {
+    origin: TransactionEventOrigin;
+    actions?: any;
```

Comment:
we can strongly type this. Please add interface `Action` like this:
```
export interface Action {
    transaction: Transaction;
    recordRef: any;
}
```
Note in `IgxTransactionService.redo` method there is `useInUndo` parameter in the action type - this is a leftover, and you can safely remove it.